src/qibolab/instruments/icarusqfpga.py

```diff
@@ -1,259 +1,453 @@
+
+    def play_sequences(
+        self,
+        qubits: Dict[QubitId, Qubit],
+        couplers,
+        sequences: List[PulseSequence],
+        options: ExecutionParameters,
+    ):
+        pass
 
-            if retval:
-                return s.recv(1024).decode("utf-8")
+    def connect(self):
+        """Currently we only connect to the board when we have to send a
+        command."""
+        # Request the version from the board
+        ver = self.device.get_server_version()
+        log.info(f"Connected to {self.name}, version: {ver}")
 
-    def play_sequence(self):
-        self.fire()
+    def start(self):
+        pass
 
-    def connect(self):
+    def stop(self):
         pass
 
     def disconnect(self):
         pass
 
-    @staticmethod
-    def _hexify(pins):
-        return int(
-            "".join(["1" if i in set(pins) else "0" for i in reversed(range(24))]), 2
-        )
-
-
-class IcarusQFPGA(Instrument):
-    """Driver for the IcarusQ RFSoC socket-based implementation."""
-
-    def __init__(self, name, address, port=8080):
-        super().__init__(name, address)
-        self._dac_sample_size = 65536
-        self._adc_sample_size = 65536
-        self._dac_nchannels = 16
-        self._adc_nchannels = 16
-        self._dac_sampling_rate = 5898240000
-        self._adc_sampling_rate = 1966080000
-        self._nshots = 0
-
-        self.port = port
-        self._thread = None
-        self._buffer = None
-        self._adcs_to_read = None
-        self.nshots = None
-
-    def setup(self, dac_sampling_rate, adcs_to_read, **kwargs):
-        """Sets the sampling rate of the RFSoC. May need to be repeated several
-        times due to multi-tile sync error.
-
-        Arguments:
-            dac_sampling_rate_id (int): Sampling rate ID to be set on the RFSoC.
-            dac_sampling_rate_6g_id (int): Optional sampling rate ID for the 6GS/s mode if used.
-        """
-        self._adcs_to_read = adcs_to_read
-        self._dac_sampling_rate = dac_sampling_rate
-
-    def translate(self, sequence: List[Pulse], nshots):
-        """Translates the pulse sequence into a numpy array."""
+    def sweep(self):
+        pass
 
-        # Waveform is 14-bit resolution on the DACs, so we first create 16-bit arrays to store the data.
-        waveform = np.zeros((self._dac_nchannels, self._dac_sample_size), dtype="i2")
 
-        # The global time can first be set as float to handle rounding errors.
-        time_array = (
-            1 / self._dac_sampling_rate * np.arange(0, self._dac_sample_size, 1)
+class RFSOC_RO(RFSOC):
+    """IcarusQ RFSoC attached with readout capability."""
+
+    available_sweep_parameters = {
+        Parameter.amplitude,
+        Parameter.duration,
+        Parameter.frequency,
+        Parameter.relative_phase,
+        Parameter.start,
+    }
+
+    def __init__(
+        self,
+        name,
+        address,
+        dac_sampling_rate=5898.24,
+        adc_sampling_rate=1966.08,
```

Comment:
These numbers are not trivial, so I'd better see them as top-level constants.

Reply:
Resolved in c08af5d